tests/conftest.py

```diff
@@ -0,0 +1,496 @@
+
+    Arguments:
+        --dj-verbose (bool):  Default True. Pass print statements from Elements.
+        --dj-teardown (bool): Default True. Delete pipeline on close.
+        --dj-datadir (str):  Default ./tests/user_data. Relative path of test CSV data.
+    """
+    parser.addoption(
+        "--dj-verbose",
+        action="store",
+        default="True",
+        help="Verbose for dj items: True or False",
+        choices=("True", "False"),
+    )
+    parser.addoption(
+        "--dj-teardown",
+        action="store",
+        default="True",
+        help="Verbose for dj items: True or False",
+        choices=("True", "False"),
+    )
+    parser.addoption(
+        "--dj-datadir",
+        action="store",
+        default="./tests/user_data",
+        help="Relative path for saving tests data",
+    )
+
+
+@pytest.fixture(autouse=True, scope="session")
+def setup(request):
+    """Take passed commandline variables, set as global"""
+    global verbose, _tear_down, test_user_data_dir, verbose_context
+
+    verbose = value_to_bool(request.config.getoption("--dj-verbose"))
+    _tear_down = value_to_bool(request.config.getoption("--dj-teardown"))
+    test_user_data_dir = Path(request.config.getoption("--dj-datadir"))
+    test_user_data_dir.mkdir(exist_ok=True)
+
+    if not verbose:
+        logging.getLogger("datajoint").setLevel(logging.CRITICAL)
+
+    verbose_context = nullcontext() if verbose else QuietStdOut()
+
+    yield verbose_context, verbose
+
+
+# --------------------  HELPER CLASS --------------------
+
+
+def null_function(*args, **kwargs):
+    pass
+
+
+# ---------------------- FIXTURES ----------------------
+
+
+@pytest.fixture(autouse=True, scope="session")
+def dj_config(setup):
+    """If dj_local_config exists, load"""
+    if pathlib.Path("./dj_local_conf.json").exists():
+        dj.config.load("./dj_local_conf.json")
+    dj.config.update(
+        {
+            "safemode": False,
+            "database.host": os.environ.get("DJ_HOST") or dj.config["database.host"],
+            "database.password": os.environ.get("DJ_PASS")
+            or dj.config["database.password"],
+            "database.user": os.environ.get("DJ_USER") or dj.config["database.user"],
+            "custom": {
+                "ephys_mode": (
+                    os.environ.get("EPHYS_MODE") or dj.config["custom"]["ephys_mode"]
+                ),
+                "database.prefix": (
+                    os.environ.get("DATABASE_PREFIX")
+                    or dj.config["custom"]["database.prefix"]
+                ),
+                "ephys_root_data_dir": (
+                    os.environ.get("EPHYS_ROOT_DATA_DIR").split(",")
+                    if os.environ.get("EPHYS_ROOT_DATA_DIR")
+                    else dj.config["custom"]["ephys_root_data_dir"]
+                ),
+            },
+        }
+    )
+    return
+
+
+@pytest.fixture(autouse=True, scope="session")
```

Comment:
```suggestion
@pytest.fixture(scope="session")
```

Can we remove `autouse=True` for this one? Sometime I'd like to run a quick test only on pipeline generation without having to download all the test data. Looks like `ingest_data` calls this fixture.